### PR TITLE
Migrate design Blueprint to an EndpointGroup [NHUB-568]

### PR DESCRIPTION
**Pull request by devketanpro**
### Purpose
Migrate design Blueprint to an async EndpointGroup

Resolves: #[NHUB-568](https://sofab.atlassian.net/browse/NHUB-568)

[NHUB-568]: https://sofab.atlassian.net/browse/NHUB-568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: newsroom/design.py [hunk 1]
```diff
@@ -39,11 +50,14 @@ async def detail():
     return await render_template("wire_item.html", item=item, previous_versions=previous_versions)
 
 
-@blueprint.route("/design/")
-async def index():
+@blueprint.endpoint("/design/", auth=False)
+async def index(args: None, params: None, req: Request):
```
Comment (eos87) [inline newsroom/design.py]: nit: you probably don't need all these arguments
```suggestion
async def index():
```

File: newsroom/design.py [hunk 2]
```diff
@@ -39,11 +50,14 @@ async def detail():
...
     return await render_template("design_index.html")
 
 
-@blueprint.route("/design/<page>")
-async def page(page):
-    return await render_template("design_%s.html" % page)
+@blueprint.endpoint("/design/<string:page>", auth=False)
+async def page(args: RouteArguments, params: None, req: Request):
+    return await render_template("design_%s.html" % args.page)
```
Comment (eos87) [inline newsroom/design.py]: nit:
```suggestion
    return await render_template(f"design_{args.page}.html")
```

Also, we might need to check if the template exists first, otherwise it will be a 500 error. But we can do it in a separate PR.

File: newsroom/design.py [hunk 3]
```diff
@@ -39,11 +50,14 @@ async def detail():
...
+@blueprint.endpoint("/design/", auth=False)
+async def index(args: None, params: None, req: Request):
     return await render_template("design_index.html")
 
 
-@blueprint.route("/design/<page>")
-async def page(page):
-    return await render_template("design_%s.html" % page)
+@blueprint.endpoint("/design/<string:page>", auth=False)
+async def page(args: RouteArguments, params: None, req: Request):
```
Comment (eos87) [inline newsroom/design.py:58]: nit:
```suggestion
async def page(args: RouteArguments):
```

File: newsroom/design.py [hunk 4]
```diff
@@ -1,14 +1,25 @@
+from typing import Optional
```
Comment (eos87) [inline newsroom/design.py]: We can use `| None` instead
```suggestion
```

File: newsroom/design.py [hunk 5]
```diff
@@ -1,14 +1,25 @@
...
+from superdesk.core.types import Request
+from superdesk.core.module import Module
 
+blueprint = EndpointGroup("design", __name__)
 
-@blueprint.route("/design/detail")
-async def detail():
+
+class RouteArguments(BaseModel):
+    page: Optional[str] = None
```
Comment (eos87) [inline newsroom/design.py]: Following above suggestion
```suggestion
    page: str | None = None
```

**Review by eos87 — APPROVED** [review]
Looks good. I just left a few silly comments.

File: newsroom/design.py (hunk 3 above)
Comment (devketanpro) [reply newsroom/design.py:58]: I think here we need all.
